time1: 0.1
time2: 0.5
time3: 0.2
